[screens][iOS][Android] Upgrade `react-native-screens@3.8.0 ➡️ 3.10.1`

bbarthec opened the pull request:
# Why

[ENG-2603](https://linear.app/expo/issue/ENG-2603/react-native-screens-380-3100)
Supersedes #15375 
Previous upgrade: #14544

# How

1. `et uvm --module "react-native-screens" --commit "3.10.1" --target "expo-go"`
2. `yarn`
3. Fixed vendored Android files:
    - add missing imports: 
      - `import host.exp.expoview.BuildConfig`
      - `import host.exp.expoview.R`
    - fix nullability issue coming from incompatible `androidx.appcompat:appcompat:1.1.0 (used in screens) ↔️ 1.3.0 (used in Expo)`

# Test Plan

- [x] Test **ncl** in **Expo Go (unversioned)** on Android & iOS
  - clicked here & there randomly
  - openeded `Screens` screen and played with available options
  - on iOS checked portrait & horizontal orientations

bbarthec committed:
et uvm --module "react-native-screens" --commit "3.9.0" --target "expo-go"

- Additionally yarn.lock was updates with
  - `@react-native-masked-view/masked-view@0.2.5 -> 0.2.6` (followup for #15362)
  - `react-native-pager-view@5.4.6 - > 5.4.9` (followup for #15348)
  - `react-native-webview@11.13.0 -> 11.15.0` (followup for #15330)

bbarthec committed:
Update CHANGELOG

bbarthec committed:
Update CHANGELOG.md

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>

bbarthec committed:
[screens][Android] Fix vendored files

Add missing imports:
- `import host.exp.expoview.BuildConfig`
- `import host.exp.expoview.R`

bbarthec committed:
[screens][Android] Run `gradlew :expoview:spotlessApply`

bbarthec committed:
Upgrade `react-native-screens@~3.8.0 ➡️ ~3.10.0`

Run `et uvm --module "react-native-screens" --commit "3.9.0" --target "expo-go"`
Run `yarn` to update `yarn.lock`

bbarthec committed:
[Android] Run `./gradlew :expoview:spotlessApply`

bbarthec committed:
[screens][Android] Fix vendored files

Add missing imports:
- `import host.exp.expoview.BuildConfig`
- `import host.exp.expoview.R`

Fix nullability issue coming from incompatible `androidx.appcompat:appcompat:1.1.0 (used in screens) ↔️ 1.3.0 (used in Expo)`

bbarthec committed:
Update CHANGELOG

bbarthec committed:
[iOS][bare-expo] Reinstall pods

Kudo reviewed (approved):
LGTM 👍

tsapeta reviewed (approved):
It would be nice to automate applying those fixes, but since vendoring on Android needs solid refactoring, I'm fine with merging this as is.

bbarthec commented:
Before merging I'm waiting for a fix for a problem that surfaces in `Test suite / android`. Hopefully `react-native-screens` team can fix it ASAP.

bbarthec committed:
Upgrade react-native-screens@~3.10.0 ➡️ ~3.10.1

Run `et uvm --module "react-native-screens" --commit "3.10.1" --target "expo-go"`
Run `yarn` to update `yarn.lock`
Run `./gradlew :expoview:spotlessApply`
Re-added missing `R` and `BuildConfig` imports

brentvatne reviewed (approved):
nice! can you also update versioned code for sdk 44?

Kudo commented:
i'll help to update versioned code together with reanimated.

bbarthec commented:
I'm merging it and backporting to SDK 44 will be done in a separate PR 🚢
